src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -1322,16 +1443,39 @@ private static unsafe nuint NarrowUtf16ToAscii_Sse2(char* pUtf16Buffer, byte* pA
             }
             else
             {
-                if ((Sse2.MoveMask(Sse2.AddSaturate(utf16VectorFirst.AsUInt16(), asciiMaskForAddSaturate).AsByte()) & NonAsciiDataSeenMask) != 0)
+                if (Sse2.IsSupported)
                 {
-                    return 0;
+                    if ((Sse2.MoveMask(Sse2.AddSaturate(utf16VectorFirst.AsUInt16(), asciiMaskForAddSaturate).AsByte()) & NonAsciiDataSeenMask) != 0)
+                    {
+                        return 0;
+                    }
+                }
+                else
+                {
+                    if (AdvSimd.Arm64.IsSupported)
+                    {
+                        if ((AdvSimd.Arm64.MaxAcross(AdvSimd.AddSaturate(utf16VectorFirst.AsUInt16(), asciiMaskForAddSaturate).AsByte()).ToScalar() & NonAsciiDataSeenMask) != 0)
```

Comment:
>.AsByte() [](start = 126, length = 9)

No need to convert to `AsByte()`. All you want to check is if there is any non-zero in the result which `MaxAcross` will do it regardless of the type of vector.